tools/regex_linter.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python
+import argparse
+import collections
+import re
+
+from clang_tidy import get_changed_files, get_changed_lines
+
+RegexLint = collections.namedtuple("RegexLint", ["error_code", "error_desc", "regex"])
+LintError = collections.namedtuple(
+    "LintError", ["regex_lint", "line_number", "column_number", "filename"]
+)
+
+LINTS = [
+    RegexLint(
+        error_code="deprecated-assert",
+        error_desc="Don't use AT_XXX checks in new code, prefer TORCH_CHECK/TORCH_INTERNAL_ASSERT/TORCH_WARN",
```

Comment:
Please also give guidance about what the different new macros mean, to help people pick the correct one.